pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,591 @@
+		pallet_prelude::*,
+		traits::Incrementable,
+	};
+	use frame_system::pallet_prelude::*;
+	use pallet_nfts::{CancelAttributesApprovalWitness, DestroyWitness, MintWitness};
+	use sp_runtime::BoundedVec;
+	use sp_std::vec::Vec;
+
+	use super::*;
+
+	/// State reads for the non-fungibles API with required input.
+	#[derive(Encode, Decode, Debug, MaxEncodedLen)]
+	#[cfg_attr(feature = "std", derive(PartialEq, Clone))]
+	#[repr(u8)]
+	#[allow(clippy::unnecessary_cast)]
+	pub enum Read<T: Config> {
+		/// Total item supply of a specified collection.
+		#[codec(index = 0)]
+		TotalSupply(CollectionIdOf<T>),
+		/// Account balance for a specified collection.
+		#[codec(index = 1)]
+		BalanceOf {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The owner of the collection .
+			owner: AccountIdOf<T>,
+		},
+		/// Allowance for an operator approved by an owner, for a specified collection or item.
+		#[codec(index = 2)]
+		Allowance {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: Option<ItemIdOf<T>>,
+			/// The owner of the collection item.
+			owner: AccountIdOf<T>,
+			/// The delegated operator of collection item.
+			operator: AccountIdOf<T>,
+		},
+		/// Owner of a specified collection item.
+		#[codec(index = 5)]
+		OwnerOf {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: ItemIdOf<T>,
+		},
+		/// Attribute value of a specified collection item. (Error: bounded collection is not
+		/// partial)
+		#[codec(index = 6)]
+		GetAttribute {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: ItemIdOf<T>,
+			/// The namespace of the attribute.
+			namespace: AttributeNamespaceOf<T>,
+			/// The key of the attribute.
+			key: BoundedVec<u8, T::KeyLimit>,
+		},
+		/// Details of a specified collection.
+		#[codec(index = 9)]
+		Collection(CollectionIdOf<T>),
+		/// Next collection ID.
+		#[codec(index = 10)]
+		NextCollectionId,
+		/// Metadata of a specified collection item.
+		#[codec(index = 11)]
+		ItemMetadata {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: ItemIdOf<T>,
+		},
+	}
+
+	/// Results of state reads for the non-fungibles API.
+	#[derive(Debug)]
+	#[cfg_attr(feature = "std", derive(PartialEq, Clone))]
+	pub enum ReadResult<T: Config> {
+		/// Total item supply of a collection.
+		TotalSupply(u128),
+		/// Account balance for a specified collection.
+		BalanceOf(u32),
+		/// Allowance for an operator approved by an owner, for a specified collection or item.
+		Allowance(bool),
+		/// Owner of a specified collection owner.
+		OwnerOf(Option<AccountIdOf<T>>),
+		/// Attribute value of a specified collection item.
+		GetAttribute(Option<Vec<u8>>),
+		/// Details of a specified collection.
+		Collection(Option<CollectionDetailsFor<T>>),
+		/// Next collection ID.
+		NextCollectionId(Option<CollectionIdOf<T>>),
```

Comment:
```suggestion
		NextCollectionId(CollectionIdOf<T>),
```
If we suppose that `T::CollectionId::initial_value() == 0`, we could probably return a value instead of `Option` from this read.